packages/wonder-blocks-form/components/checkbox-core.js

```diff
@@ -0,0 +1,195 @@
+// @flow
+// The internal stateless ☑️ Checkbox.
+
+import * as React from "react";
+import {StyleSheet} from "aphrodite";
+
+import Color, {mix, fade} from "@khanacademy/wonder-blocks-color";
+import {View, addStyle} from "@khanacademy/wonder-blocks-core";
+import Icon from "@khanacademy/wonder-blocks-icon";
+
+import type {ClickableHandlers} from "@khanacademy/wonder-blocks-core";
+import type {IconAsset} from "@khanacademy/wonder-blocks-icon";
+
+import type {ChoiceCoreProps} from "../util/types.js";
+
+type Props = {|
+    ...ChoiceCoreProps,
+    ...ClickableHandlers,
+    hovered: boolean,
+    focused: boolean,
+    pressed: boolean,
+|};
+
+const {blue, red, white, offWhite, offBlack16, offBlack32, offBlack50} = Color;
+
+const StyledInput = addStyle("input");
+
+const checkboxCheck: IconAsset = {
+    small:
+        "M11.263 4.324a1 1 0 1 1 1.474 1.352l-5.5 6a1 1 0 0 1-1.505-.036l-2.5-3a1 1 0 1 1 1.536-1.28L6.536 9.48l4.727-5.157z",
+};
+
+export default class CheckboxCore extends React.Component<Props> {
+    render() {
+        const {
+            checked,
+            disabled,
+            error,
+            id,
```

Comment:
It seems like there are two routes we can go.  We can use the `for`/`id` pattern with `<label>`s or `aria-labelledby`/`id` pattern.  My preference would be for using `aria-labelledby`.  The reason being that we using `aria-labelledby` in other places and it would be nice to have fewer a11y patterns to think about.  cc @somewhatabstract

Reply:
I did some digging and according https://www.w3.org/WAI/tutorials/forms/labels/ we should go with `<label>` and then use `for` and `id`.